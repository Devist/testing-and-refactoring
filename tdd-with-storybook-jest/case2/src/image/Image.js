import React from 'react';

import './image.css'
import noImage from '../img/no_image.png'

export default function Image(props) {
  const { image = noImage } = props
  return (
    <div className='image-container'>
      <img src={image} alt={image} />
    </div>
  )
}