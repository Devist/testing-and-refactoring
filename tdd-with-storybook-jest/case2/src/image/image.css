.image-container {
    width: 400px;
    margin: 0 auto;
}

.image-container img {
    width: 100%;
}