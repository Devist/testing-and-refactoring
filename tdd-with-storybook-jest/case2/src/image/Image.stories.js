import React from 'react'

import Image from './Image'
import image1 from '../img/vintage_clock_1.png'

export const Default = () => <Image image={image1} />

export const EmptyImage = () => <Image />

export default {
    title: 'Case2[solution]/Image',
    component: Default,
};