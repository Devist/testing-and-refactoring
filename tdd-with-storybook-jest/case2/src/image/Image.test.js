import React from 'react';
import renderer from 'react-test-renderer';
import Image from './Image';

import image from '../img/vintage_clock_1.png'

test('제품 이미지(snampshots) - 존재하지 않는다면 No-Image가 표현된다.', () => {
  const component = renderer.create(
    <Image />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('제품 이미지(snampshots) - 빈티지 시계가 표현된다.', () => {
    const component = renderer.create(
    <Image image={image}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});