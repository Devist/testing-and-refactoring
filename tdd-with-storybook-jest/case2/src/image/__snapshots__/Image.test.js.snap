// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`제품 이미지(snampshots) - 빈티지 시계가 표현된다. 1`] = `
<div
  className="image-container"
>
  <img
    alt="vintage_clock_1.png"
    src="vintage_clock_1.png"
  />
</div>
`;

exports[`제품 이미지(snampshots) - 존재하지 않는다면 No-Image가 표현된다. 1`] = `
<div
  className="image-container"
>
  <img
    alt="no_image.png"
    src="no_image.png"
  />
</div>
`;
